keras/src/export/litert.py

```diff
@@ -0,0 +1,548 @@
+                "Skipping AOT compilation."
+            )
+
+        if self.verbose:
+            io_utils.print_msg(
+                f"LiteRT export completed. Base model: {filepath}"
+            )
+            if compiled_models:
+                io_utils.print_msg(
+                    f"AOT compiled models: {len(compiled_models.models)} "
+                    "variants"
+                )
+
+        return compiled_models if compiled_models else filepath
+
+    def _ensure_model_built(self):
+        """
+        Ensures the model is built before conversion.
+
+        For models that are not yet built, this attempts to build them
+        using the input signature or model.inputs.
+        """
+        if self.model.built:
+            return
+
+        if self.verbose:
+            io_utils.print_msg("Building model before conversion...")
+
+        try:
+            # Try to build using input_signature if available
+            if self.input_signature:
+                input_shapes = tree.map_structure(
+                    lambda spec: spec.shape, self.input_signature
+                )
+                self.model.build(input_shapes)
+            # Fall back to model.inputs for Functional/Sequential models
+            elif hasattr(self.model, "inputs") and self.model.inputs:
+                input_shapes = [inp.shape for inp in self.model.inputs]
+                if len(input_shapes) == 1:
+                    self.model.build(input_shapes[0])
+                else:
+                    self.model.build(input_shapes)
+            else:
+                raise ValueError(
+                    "Cannot export model to the litert format as the "
+                    "input_signature could not be inferred. Either pass an "
+                    "`input_signature` to `model.export()` or ensure that the "
+                    "model is already built (called once on real inputs)."
+                )
+
+            if self.verbose:
+                io_utils.print_msg("Model built successfully.")
+
+        except Exception as e:
+            if self.verbose:
+                io_utils.print_msg(f"Error building model: {e}")
+            raise ValueError(
+                f"Failed to build model: {e}. Please ensure the model is "
+                "properly defined or provide an input_signature."
+            )
+
+    def _convert_to_tflite(self, input_signature):
+        """Converts the Keras model to TFLite format.
+
+        Returns:
+            A bytes object containing the serialized TFLite model.
+        """
+        is_sequential = isinstance(self.model, tf.keras.Sequential)
+
+        # Try direct conversion first for all models
+        try:
+            if self.verbose:
+                model_type = "Sequential" if is_sequential else "Functional"
+                io_utils.print_msg(
+                    f"{model_type} model detected. Trying direct conversion..."
+                )
```

Comment:
Remove. Keras in general has almost no console output in the success case. I don't think this message brings any value to the user.